src/python/pants/engine/target.py

```diff
@@ -807,25 +816,38 @@ class methods:
     """
 
     required_fields: ClassVar[tuple[type[Field], ...]]
+    unconditional_opt_out: ClassVar[type[Field] | None] = None
 
     address: Address
 
+    @classmethod
+    def conditional_opt_out(cls, tgt: Target) -> bool:
+        """If `True` is returned, the target will not match with the field set, even if it has the
```

Comment:
```suggestion
        """If True the target will not match with the field set, even if it has the
```

May help fit it on one line.